### ISSUE TITLE
Add type hinting

### ISSUE THREAD

**Issue by sandalwoodbox** [projectfluent/python-fluent#165]
Currently, mypy checking of my python code fails with:

```
 error: Skipping analyzing 'fluent.runtime': found module but no type hints or library stubs
```

I'll just ignore the error for now but it would be nice to have type hint support. It looks like this is blocked by https://github.com/projectfluent/python-fluent/issues/162.

**Comment by Pike** [comment]
Also a port of https://github.com/projectfluent/fluent.js/issues/524, I'd say.
